crates/api/src/community.rs

```diff
@@ -591,6 +594,18 @@ impl Perform for BanFromCommunity {
       if blocking(context.pool(), ban).await?.is_err() {
         return Err(APIError::err("community_user_already_banned").into());
       }
+
+      // Also unsubscribe them from the community, if they are subscribed
+      let community_follower_form = CommunityFollowerForm {
+        community_id: data.community_id,
+        user_id: banned_user_id,
+        pending: false,
+      };
+      blocking(context.pool(), move |conn: &'_ _| {
+        CommunityFollower::unfollow(conn, &community_follower_form)
+      })
+      .await?
+      .ok();
```

Comment:
They might not be a follower, so the `.ok()` should ignore an error.

Reply:
Worth noting that this is not gonna affect federated users. So if two users from the same instance follow our community and we ban one, the other instance wont know about that and we keep delivering community activities to the instance's shared_inbox. And it will show those to the banned user. We would have to implement a new activity to handle that.

Reply:
Def needs to be a priority soon, getting more moderation-related activities federating.